frontend/src/components/tag-pill/index.js

```diff
@@ -60,7 +64,10 @@ export default class TagPillGroup extends React.Component {
           <TagPill
             className={pillClassName}
             selected={
-              this.state.selected === tag.id || this.state.selected === index
+              this.props.onChange
+                ? tag.id === this.props.value || index === this.props.value
```

Comment:
There's quite a lot of logic in this one statement here.. could you put it inside a function instead so it would be easier to read what is happening here?